dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/ClientBase.cs

```diff
@@ -266,8 +262,8 @@ private static CompletionsOptions CreateCompletionsOptions(string text, Complete
             FrequencyPenalty = (float?)requestSettings.FrequencyPenalty,
             PresencePenalty = (float?)requestSettings.PresencePenalty,
             Echo = false,
-            ChoicesPerPrompt = 1,
-            GenerationSampleCount = 1,
+            ChoicesPerPrompt = requestSettings.ResultsPerPrompt,
```

Comment:
FYI: Has a valid range of 1 to 128. https://learn.microsoft.com/en-us/dotnet/api/azure.ai.openai.completionsoptions.choicesperprompt?view=azure-dotnet-preview

Reply:
Good catch, will add this validation. Interestingly enough current SDK impl don't validate that size neither the usage of ChoicesPerPrompt > GenerationSampleCount which gives a bad request.